### PR TITLE
Adds Shimmer Loading Effect

### LINKED ISSUE #121: Create Loading state for Claim Tracker
### Task
We need to define a loading state for the page, in case the internet connection is slow.

### Details / Context
The loading state should work on mobile and desktop, no matter the browser or device. It should be the first thing to load on a page.

### Acceptance Criteria

- [x] Style designed for desktop and mobile

### PR DESCRIPTION
**Pull request by lomky**
### Description

  - Adds a shimmer component & styling for it
  - Adds shimmerable components for:
    - Key&value fields
    - Text lines
    - Next Steps block
  - Embeds those components in the Claim Section where appropriate
  - Passes loading state down through the component stack from where we make the API call
    - **nb:** its possible there's a more elegant way to do this, but I'm calling Good Enough for now with our looming deadlines
  - Adds stories for each

### Changes
Component | Normal | Loading
-- | -- | --
Main | main app normal | main app loading
Claim Details | image | image
Claim Status | image | image
NextSteps | image | image
InfoField | image | image
TextLine | image | image
Shimmer | - | image
ShimmerField | - | image

### Design Review
The Storybook for Design's review can be accessed [here.](https://60705d04dcad7600211e34d2-kfvehegept.chromatic.com)
See the Page Section Main component, and its Loading state.

### Testing
(tests should pass after #219 merges)
The loading state cannot be invoked naturally the app until we split the page load into two. For now

  - `git fetch && git checkout kat/121_shimmer_loading && git pull`
  - `yarn dev`
  - Navigate to http://localhost:3000/claimstatus - App should load without issue.
  - edit `pages/index.tsx` line 37 to  `loading={true}`
  - Navigate to http://localhost:3000/claimstatus - App should load with the shimmers.

You can also examine the various componenets & page sections listed above in Storybook and toggle Loading to see each one.
  - `git fetch && git checkout kat/121_shimmer_loading && git pull`
  - `yarn storybook`
  
Resolves #121 

- [ ] Claim Status shows a shimmer for pending data
- [ ] Ticket filed for splitting page load

### CODE REVIEW

File: tests/components/main.test.tsx [hunk 1]
```diff
@@ -15,21 +15,36 @@ describe('Main component shows the page', () => {
...
     expect(screen.queryByText('Claim Status')).toBeInTheDocument()
   })
 })
 
+describe('Main component shows loading', () => {
+  it('has our title, details, and body with shimmer', () => {
+    const mockRouter = {
+      locale: 'en',
+    }
+    ;(useRouter as jest.Mock).mockReturnValue(mockRouter)
```
Comment (kalvinwang) [inline tests/components/main.test.tsx:30]: Do you know why these lines begin with a semicolon? I see the pattern exists below too.

**Review by kalvinwang — APPROVED** [review]
Code lgtm, and cool feature! Were you able to test that it looks good on mobile (via dev tools)? Otherwise no concerns!

File: tests/components/main.test.tsx (hunk 1 above)
Comment (lomky) [reply tests/components/main.test.tsx:30]: I do not remember! I'm duping the pattern above and I don't recall why it looks like this

**Comment by lomky** [comment]
oh good call! the labels overlap - will make them relative 
image

**Comment by lomky** [comment]
Fixed:
Normal:
image

Mobile:
image

File: tests/components/main.test.tsx (hunk 1 above)
Comment (lomky) [reply tests/components/main.test.tsx:30]: taking away the semicolon gives you the error:

```
[tsserver 2349] [E] This expression is not callable.                                                                                   
   Type '{ locale: string; }' has no call signatures.
```

which makes me suspect this is Jest Mocks specific syntax magic

**Comment by rocketnova** [comment]
@lomky I'm seeing a warning on Github Actions that looks like it might be related to this PR. I think we need a return type on `export const ShimmerField`:

Screen Shot 2021-06-14 at 2 51 26 PM

Should we create a new ticket for that?

**Comment by lomky** [comment]
@rocketnova I see that on other PRs too! I wonder why it didn't show up on this one?

I'll see if I can't fix it in ~15m and submit a PR directly. If not I'll file an issue

**Pull request merged**